onnx/defs/generator/defs.cc

```diff
@@ -225,3 +225,56 @@ TensorProto message, and be valid as an output type.
         }
         propagateShapeFromInputToOutput(ctx, 0, 0);
     });
+
+ONNX_OPERATOR_SCHEMA(Multinomial)
+    .SetDoc(R"DOC(
+Generate a tensor of samples from a multinomial distribution according to the probabilities
+of each of the possible outcomes.
+)DOC")
+    .Attr(
+        "sample_size",
+        "Number of times to sample.",
+        AttributeProto::INT,
+        static_cast<int64_t>(0))
+    .Attr(
+        "seed",
+        "(Optional) Seed to the random generator, if not specified we will auto generate one.",
+        AttributeProto::FLOAT,
+        OPTIONAL)
+    .Input(
+        0,
+        "input",
+        "Input tensor with shape [batch_size, class_size], where class_size is the number of all possible outcomes.",
+        "T1")
+    .Output(
+        0,
+        "output",
+        "Output tensor with shape [batch_size, sample_size], where sample_size is the number of times to sample. ",
+        "T2")
+    .TypeConstraint(
+        "T1",
+        {"tensor(float16)", "tensor(float)", "tensor(double)"},
+        "Constrain input types to float tensors.")
+    .TypeConstraint(
+        "T2",
+        {"tensor(int32)", "tensor(int64)"},
```

Comment:
How is the backend supposed to know which output type it should take?

Reply:
Yes, this is not clear to me either. If we want both, we probably should add a "dtype" attribute that specifies the output type

Reply:
As for the backend test case, I've looked at it and realized that this is a generator op. Although I think one could implement an AB testing comparing the result of a random function against another (say std::mt19937), it would be too stringent to assume that all backend implementation must conform to a specific algorithm.